### ISSUE TITLE
Multi-Languange Component: rootNamespace is ignored when generating C# SDK

### ISSUE THREAD

**Issue by MitchellGerdisch** [pulumi/pulumi#9701]
### What happened?

As per https://www.pulumi.com/docs/guides/pulumi-packages/schema/#packagelanguage 
there is a `rootNamespace` property for C# that can be set to override the default `Pulumi.XXXX` namespace naming convention.
However, setting this value has no effect.

### Steps to reproduce

1. Clone https://github.com/pulumi/pulumi-aws-quickstart-vpc
2. Update `schema.yaml` to have (note the addition of `rootNamespace: MyNamespace`
```
language:
    csharp:
        packageReferences:
            Pulumi: 3.*
            Pulumi.Aws: 4.*
        namespaces:
            aws-quickstart-vpc: AwsQuickStartVpc
        rootNamespace: MyNamespace
```
3. Run `make gen_dotnet_sdk `
4. Confirm that the dotnet SDK was recreated.
5. Look at the generated `Provider.cs` and see that the namespace is still `namespace Pulumi.AwsQuickStartVpc` and doesn't use the custom `MyNamespace` value.

### Expected Behavior

The specified `rootNamespace` should be used in generating the dotnet SDK

### Actual Behavior

The generated dotnet SDK has a namespace that starts with `Pulumi` instead of the specified `rootNamespace`

### Versions used

CLI          
Version      3.33.1
Go Version   go1.17.10
Go Compiler  gc

Plugins
NAME  VERSION
go    unknown

Host     
OS       darwin
Version  12.3.1
Arch     x86_64

This project is written in go (/usr/local/bin/go v1.17 darwin/amd64)

Current Stack: dev

Found no resources associated with dev

Found no pending operations associated with dev

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by AaronFriel** [comment]
Thanks @MitchellGerdisch, @iwahbe is looking into this!

**Comment by iwahbe** [comment]
The `rootNamespace` facility was added in #8735 and released as part of [v3.23.0](https://github.com/pulumi/pulumi/releases/tag/v3.23.0). The https://github.com/pulumi/pulumi-aws-quickstart-vpc uses v3.13.2. Upgrading fixed the issue. 

I filed https://github.com/pulumi/pulumi-aws-quickstart-vpc/issues/22 in response.
